demosplan/DemosPlanCoreBundle/Repository/NotificationReceiverRepository.php

```diff
@@ -99,7 +99,7 @@ public function generateObjectValues($entity, array $data)
             $entity->setLabel($data['label']);
         }
         if (array_key_exists('procedureId', $data)) {
-            $entity->setProcedureId($data['procedureId']);
+            $entity->setProcedure($data['procedureId']);
```

Comment:
What is stored in `$data['procedureId']`? Reading the key I would guess there is an ID stored, which is a string.
`setProcedure()` expects a `ProcedureInterface` (mentioned in the docblock)
To solve this, it would be necessary to find all usages and search for actual incoming type(s).

Reply:
done in [685364f](https://github.com/demos-europe/demosplan-core/pull/2960/commits/685364f97784d85dd520137c6bcee9f67c93e07c)

Reply:
I'm pretty sure `find()` can actually also return `null` in case of there is no related Entity to the given ID. I don't know why this isn't defined at the `find()`-method.
`NotificationReceiver::setProcedure()` expects an `ProcedureInterface` but never null. (related property is nullable=false)
To be correct, here should be thrown an exception in case of `find()` returns null.
In other words: This case isn't solvable by a simple type declaration.

Reply:
done in [f8f6053](https://github.com/demos-europe/demosplan-core/pull/2960/commits/f8f60532b56a758a0da95c3d345b88314f48427f)